### ISSUE TITLE
Does not works with maven 3.3.3

### ISSUE THREAD

**Issue by torsten-liermann** [release-engineering/pom-manipulation-ext#226]
Installation the pom-manipulation-ext.jar into project folder .mvn via extension.xml will produce with maven 3.3.3 the exception:

    java.lang.ClassCastException: org.codehaus.plexus.util.xml.Xpp3Dom cannot be cast to org.codehaus.plexus.util.xml.Xpp3Dom

**Comment by rnc** [comment]
Using
```
<?xml version="1.0" encoding="UTF-8"?>
<extensions>
<extension>
  <groupId>org.commonjava.maven.ext</groupId>
  <artifactId>pom-manipulation-ext</artifactId>
  <version>0.7.2</version>
</extension>
</extensions>
```

and version 0.7.2 I am not seeing a ClassCastException. Can give any logs/test case?

**Comment by torsten-liermann** [comment]
I have the same extensions.xml:

```
<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
  <extension>
    <groupId>
            org.commonjava.maven.ext
    </groupId>
    <artifactId>pom-manipulation-ext</artifactId>
    <version>1.7.2</version>
  </extension>
</extensions>
```

The log output:
```
[main] INFO  o.c.maven.ext.manip.io.PomIO - de.zeiss.forum:EmrXml:4.0.2-SNAPSHOT [pom=D:\dev\lehel\temp\ForumServer-4.0.2\EmrXml\EmrXml\pom.xml] modified! Rewri
ting.
[main] INFO  o.c.maven.ext.manip.io.PomIO - Rewriting: de.zeiss.forum:EmrXml:ejb:[inherited] in place of: de.zeiss.forum:EmrXml:ejb:[inherited]
       to POM: D:\dev\lehel\temp\ForumServer-4.0.2\EmrXml\EmrXml\pom.xml
[main] ERROR o.c.m.ext.manip.ManipulatingEventSpy - Extension failure
java.lang.ClassCastException: org.codehaus.plexus.util.xml.Xpp3Dom cannot be cast to org.codehaus.plexus.util.xml.Xpp3Dom
        at org.apache.maven.model.io.jdom.MavenJDOMWriter.updatePlugin(MavenJDOMWriter.java:1815) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.model.io.jdom.MavenJDOMWriter.iteratePlugin(MavenJDOMWriter.java:579) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.model.io.jdom.MavenJDOMWriter.updatePluginManagement(MavenJDOMWriter.java:1901) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.model.io.jdom.MavenJDOMWriter.updateBuild(MavenJDOMWriter.java:1090) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.model.io.jdom.MavenJDOMWriter.updateModel(MavenJDOMWriter.java:1630) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.model.io.jdom.MavenJDOMWriter.update(MavenJDOMWriter.java:2255) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.model.io.jdom.MavenJDOMWriter.update(MavenJDOMWriter.java:80) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.io.util.AbstractJDOMWriter.write(AbstractJDOMWriter.java:85) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.io.util.AbstractJDOMWriter.write(AbstractJDOMWriter.java:123) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.io.util.AbstractJDOMWriter.write(AbstractJDOMWriter.java:104) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.commonjava.maven.ext.manip.io.PomIO.write(PomIO.java:183) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.commonjava.maven.ext.manip.io.PomIO.rewritePOMs(PomIO.java:161) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.commonjava.maven.ext.manip.ManipulationManager.scanAndApply(ManipulationManager.java:141) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.commonjava.maven.ext.manip.ManipulatingEventSpy.onEvent(ManipulatingEventSpy.java:112) ~[pom-manipulation-ext-1.7.2.jar:1.7.2]
        at org.apache.maven.eventspy.internal.EventSpyDispatcher.onEvent(EventSpyDispatcher.java:108) [maven-core-3.3.3.jar:3.3.3]
        at org.apache.maven.eventspy.internal.EventSpyExecutionListener.projectDiscoveryStarted(EventSpyExecutionListener.java:47) [maven-core-3.3.3.jar:3.3.3]
        at org.apache.maven.lifecycle.internal.DefaultExecutionEventCatapult.fire(DefaultExecutionEventCatapult.java:57) [maven-core-3.3.3.jar:3.3.3]
        at org.apache.maven.lifecycle.internal.DefaultExecutionEventCatapult.fire(DefaultExecutionEventCatapult.java:42) [maven-core-3.3.3.jar:3.3.3]
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:217) [maven-core-3.3.3.jar:3.3.3]
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193) [maven-core-3.3.3.jar:3.3.3]
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106) [maven-core-3.3.3.jar:3.3.3]
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862) [maven-embedder-3.3.3.jar:3.3.3]
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286) [maven-embedder-3.3.3.jar:3.3.3]
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197) [maven-embedder-3.3.3.jar:3.3.3]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_55]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_55]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_55]
        at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_55]
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289) [plexus-classworlds-2.5.2.jar:na]
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229) [plexus-classworlds-2.5.2.jar:na]
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415) [plexus-classworlds-2.5.2.jar:na]
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356) [plexus-classworlds-2.5.2.jar:na]
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-readable POM D:\dev\lehel\temp\ForumServer-4.0.2\EmrXml\EmrXml\pom.xml: input contained no data @
 @
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project  (D:\dev\lehel\temp\ForumServer-4.0.2\EmrXml\EmrXml\pom.xml) has 1 error
[ERROR]     Non-readable POM D:\dev\lehel\temp\ForumServer-4.0.2\EmrXml\EmrXml\pom.xml: input contained no data
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

After that, the pom has 0 bytes.
Without the manipulation extension, a maven 3.3.3 build runs without warnings.

**Comment by jdcasey** [comment]
It looks like the solution might be to stop shading plexus-utils into PME

**Comment by rnc** [comment]
*Arg* my fault I typed my extension.xml wrong. I can reproduce it here now :-)

**Comment by jdcasey** [comment]
Without looking at how Maven may have changed, I notice that PME includes compile-scoped deps to other modules from this project. They have not been shifted to provided scope, in spite of being shaded into the output jar file. If Maven 3.3.x switched to resolving the transitive closure for an extension using its POM, and constructing a classloader for it, you could wind up with a conflict between the plexus-utils loaded in the main Maven classloader and the one loaded transitively for maven3-model-jdom-support (used by pom-manipulation-io).

**Comment by rnc** [comment]
Workaround : put it in the ext/lib directory for now without the extension.xml

**Comment by jdcasey** [comment]
For next release, we should figure out why the other modules aren't being switched to provided scope when they get shaded in...all deps should be switched over to provided scope, to prevent inclusion of anything transitive.

**Comment by torsten-liermann** [comment]
How I can test the fix when merged?

**Comment by rnc** [comment]
Until @jdcasey does a new release it should be possible to test by building PME locally as the snapshot version and having it in your extension.xml:
```
<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
  <extension>
    <groupId>org.commonjava.maven.ext</groupId>
    <artifactId>pom-manipulation-ext</artifactId>
  <version>1.8-SNAPSHOT</version>
 </extension>
</extensions>
```

**Comment by torsten-liermann** [comment]
Thanks, 1.8-SNAPSHOT works for me.

**Comment by rnc** [comment]
Resolved by #227
